### PR TITLE
added fileds to AaveV2 subgraph on mainnet for rewards calculation

### PR DESCRIPTION
**Pull request by kuririn1**
Added aEmissionPerSecond and totalLiquidity. Token prices will be pulled off-chain.

TotalLiqudity will be only correct when data will be pulled from the beginning of the pool (it's calculated based on each mint and burn of a token in the pool).

Formula to calculate the reward rate:
Reward_APR = (AAVE_price * AEmissions_per_second * 60 * 60 * 24 * 365) / (Total_Liquidity * Pool_Token_Price * 10^18)

### CODE REVIEW

**Comment by uwe** [comment]
Is that the reserve figure that you calculate with `AaveV2Reserve`?

Screenshot 2021-12-17 at 23 33 43

I did reach it by going to the token https://etherscan.io/address/0xFFC97d72E13E01096502Cb8Eb52dEe56f74DAD7B#readProxyContract and then `POOL`. The `asset` is `UNDERLYING_ASSET_ADDRESS`. Please check, if it is possible to get the reserve data from the chain.

**Comment by kuririn1** [comment]
> Is that the reserve figure that you calculate with `AaveV2Reserve`?
> 
> Screenshot 2021-12-17 at 23 33 43
> 
> I did reach it by going to the token https://etherscan.io/address/0xFFC97d72E13E01096502Cb8Eb52dEe56f74DAD7B#readProxyContract and then `POOL`. The `asset` is `UNDERLYING_ASSET_ADDRESS`. Please check, if it is possible to get the reserve data from the chain.

No, it's not this. I checked that already. I use "reserve" name, because Aave subgraph also used this name, but there is no relation to this call.

**Comment by kuririn1** [comment]
> Is that the reserve figure that you calculate with `AaveV2Reserve`?
> 
> Screenshot 2021-12-17 at 23 33 43
> 
> I did reach it by going to the token https://etherscan.io/address/0xFFC97d72E13E01096502Cb8Eb52dEe56f74DAD7B#readProxyContract and then `POOL`. The `asset` is `UNDERLYING_ASSET_ADDRESS`. Please check, if it is possible to get the reserve data from the chain.

I don't think this is the same, because Aave subgraph would use it, but I will check to be sure.

**Comment by kuririn1** [comment]
> Is that the reserve figure that you calculate with `AaveV2Reserve`?
> 
> Screenshot 2021-12-17 at 23 33 43
> 
> I did reach it by going to the token https://etherscan.io/address/0xFFC97d72E13E01096502Cb8Eb52dEe56f74DAD7B#readProxyContract and then `POOL`. The `asset` is `UNDERLYING_ASSET_ADDRESS`. Please check, if it is possible to get the reserve data from the chain.

Confirmed. This call is showing available liquidity and it's not the same thing as total liquidity.

File: src/AaveV2/mappings/Token.ts [hunk 1]
```diff
@@ -1,7 +1,8 @@
 import { log, Address, Bytes, BigInt } from "@graphprotocol/graph-ts";
 import { AaveV2Token, Burn as BurnEvent, Mint as MintEvent } from "../../../generated/AaveV2TokenaDAI/AaveV2Token";
-import { AaveV2TokenData } from "../../../generated/schema";
+import { AaveV2TokenData, AaveV2Reserve } from "../../../generated/schema";
 import { AaveV2LendingPoolAddressesProvider } from "../../../generated/AaveV2TokenaDAI/AaveV2LendingPoolAddressesProvider";
+import { AaveV2IncentivesController } from "../../../generated/AaveV2TokenaDAI/AaveV2IncentivesController"; // ADD ABI FOR IT!!!
```
Comment (uwe) [inline src/AaveV2/mappings/Token.ts:5]: Please remove that comment (the ABI is in the PR).

File: src/AaveV2/mappings/Token.ts [hunk 2]
```diff
@@ -83,13 +89,44 @@ function handleAaveV2Token(
     }
   }
 
+  //get aEmissionsPerSecond
+  let tried_getIncentivesController = tokenContract.try_getIncentivesController();
+  if (tried_getIncentivesController.reverted) log.error("tokenContract at {} call getIncentivesController() reverted", [ tokenContract._address.toHex() ]);
```
Comment (uwe) [inline src/AaveV2/mappings/Token.ts:94]: I would prefer `{` blocks everywhere.

File: src/AaveV2/mappings/Token.ts [hunk 3]
```diff
@@ -83,13 +89,44 @@ function handleAaveV2Token(
...
 
+  //get aEmissionsPerSecond
+  let tried_getIncentivesController = tokenContract.try_getIncentivesController();
+  if (tried_getIncentivesController.reverted) log.error("tokenContract at {} call getIncentivesController() reverted", [ tokenContract._address.toHex() ]); 
+  else {
+    let incentivesControllerAddress = tried_getIncentivesController.value;
+    let incentivesControllerContract = AaveV2IncentivesController.bind(incentivesControllerAddress);
+    let tried_assets = incentivesControllerContract.try_assets(address);
+
+    if (tried_assets.reverted) log.error("AaveIncentivesController at {} call assets({}) reverted", [ incentivesControllerContract._address.toHex(), underlyingAssetAddr.toHex() ]);
```
Comment (uwe) [inline src/AaveV2/mappings/Token.ts:100]: here as well

**Review by uwe — APPROVED** [review]
Please run `yarn prettier` and do the mentioned small changes. PR is good otherwise. Thanks for your contribution.

File: src/AaveV2/mappings/Token.ts (hunk 2 above)
Comment (kuririn1) [reply src/AaveV2/mappings/Token.ts:94]: sure, me as well - I was copying the convention that was used in the previous code - I will add the block there too

**Pull request merged**